lib/postgres/postgres-pr/connection.rb

```diff
@@ -129,12 +129,97 @@ def peerport
     @conn.peerport
   end
 
+  def peerinfo
+    "#{peerhost}:#{peerport}"
+  end
+
   def current_database
     @params['database']
   end
 
-  def peerinfo
-    "#{peerhost}:#{peerport}"
+  # List of supported PostgreSQL platforms & architectures:
+  # https://postgrespro.com/docs/postgresql/16/supported-platforms
+  def map_compile_os_to_platform(compile_os)
+    return Msf::Platform::Unknown.realname if compile_os.blank?
+
+    compile_os = compile_os.downcase.encode(::Encoding::BINARY)
+
+    if compile_os.match?('linux')
+      platform = Msf::Platform::Linux
+    elsif compile_os.match?(/(darwin|mac|osx)/)
+      platform = Msf::Platform::OSX
+    elsif compile_os.match?('win')
+      platform = Msf::Platform::Windows
+    elsif compile_os.match?('free')
+      platform = Msf::Platform::FreeBSD
+    elsif compile_os.match?('net')
+      platform = Msf::Platform::NetBSD
+    elsif compile_os.match?('open')
+      platform = Msf::Platform::OpenBSD
+    elsif compile_os.match?('solaris')
+      platform = Msf::Platform::Solaris
+    elsif compile_os.match?('aix')
+      platform = Msf::Platform::AIX
+    elsif compile_os.match?('hpux')
+      platform = Msf::Platform::HPUX
+    elsif compile_os.match?('irix')
+      platform = Msf::Platform::Irix
+    else
+      # Return the query result if the value can't be mapped
+      return compile_os
+    end
+
+    platform.realname
+  end
+
+  # List of supported PostgreSQL platforms & architectures:
+  # https://postgrespro.com/docs/postgresql/16/supported-platforms
+  def map_compile_arch_to_architecture(compile_arch)
+    return '' if compile_arch.blank?
+
+    compile_arch = compile_arch.downcase.encode(::Encoding::BINARY)
+
+    if compile_arch.match?('sparc')
+      if compile_arch.include?('64')
+        arch = ARCH_SPARC64
+      else
+        arch = ARCH_SPARC
+      end
+    elsif compile_arch.include?('mips')
+      arch = ARCH_MIPS
+    elsif compile_arch.include?('ppc')
+      arch = ARCH_PPC
+    elsif compile_arch.match?('arm')
+      arch = ARCH_AARCH64
+    elsif compile_arch.match?('64')
+      arch = ARCH_X86_64
+    elsif compile_arch.match?('86') || compile_arch.match?('i686')
+      arch = ARCH_X86
+    else
+      # Return the query result if the value can't be mapped
+      arch = compile_arch
+    end
+
+    arch
+  end
+
+  # @return [Hash] Detect the platform and architecture of the MySQL server:
+  #  * :arch [String] The server architecture.
+  #  * :platform [String] The server platform.
+  def detect_platform_and_arch
+    result = {}
+
+    query_result = query('select version()').rows.join.match(/.*on (\w+-\w+-\w+-\w+)/).captures
+    server_var = query_result.join.split('-', 2)
+    server_vars = {
+      'version_compile_machine' => server_var[0],
+      'version_compile_os' => server_var[1]
+    }
```

Comment:
Genuinely not sure of the possible permutations for the version result, would it be possible to use named capture groups in the regex vs captures -> join -> split?

Not a big deal just prefer to avoid hard coded index access it the regex could provide a more flexible match.